### PR TITLE
Allow configure-web-proxy to use a custom port

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/162611

### CODE REVIEW

File: script/configure-web-proxy [hunk 1]
```diff
@@ -9,20 +9,26 @@ Options:
  -h, --help         display this help
  -p, --proxy=PROXY  web proxy to configure (default: apache)
                     choose from: nginx, apache
+ -P, --port=PORT    port to configure (default: 80)
 EOF
     exit "$1"
 }
 
-opts=$(getopt -o hp: -l help -l proxy: -n "$0" -- "$@") || usage 1
+opts=$(getopt -o hp: -l help -l proxy -l port: -n "$0" -- "$@") || usage 1
```
[the remark this answers is not in the document]
Comment (kalikiana) [reply script/configure-web-proxy]: Apparently it needs to be `hp:P` although I don't understand why.

**Review by Martchus — COMMENTED** [review]
I would prefer if the script left `nginx.conf`/`listen.conf` untouched if no port is specified.

**Comment by perlpunk** [comment]
I don't understand the usage anymore right now.
I checked out your branch and ran some tests:
```
# bad
% ./script/configure-web-proxy  -P 8080
$web_proxy=apache $web_port=--
% ../script/configure-web-proxy  -p nginx -P 8080
$web_proxy=nginx $web_port=--
# good
% ../script/configure-web-proxy  --port 8080
$web_proxy=apache $web_port=8080
% ../script/configure-web-proxy  -p nginx --port 8080
$web_proxy=nginx $web_port=8080

File: script/configure-web-proxy (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalikiana) [reply script/configure-web-proxy]: Command: `sudo ./script/configure-web-proxy -P 9090`
getopt: `getopt -o hp:P -l help,proxy:,port: -n "$0" -- "$@"`
Parsed: `opts -P -- '9090'`

According to the docs a `:` after a short or long option means it takes a mandatory argument. I was missing the `:` after `p:`.

**Comment by kalikiana** [comment]
> I would prefer if the script left `nginx.conf`/`listen.conf` untouched if no port is specified.

That would lead to surprises when running the script more than once unless we start parsing the original file.

File: script/configure-web-proxy [hunk 2]
```diff
@@ -9,19 +9,25 @@ Options:
  -h, --help         display this help
  -p, --proxy=PROXY  web proxy to configure (default: apache)
                     choose from: nginx, apache
+ -P, --port=PORT    port to configure (default: 80)
 EOF
     exit "$1"
 }
 
-opts=$(getopt -o hp: -l help -l proxy: -n "$0" -- "$@") || usage 1
+opts=$(getopt -o hp:P -l help,proxy:,port: -n "$0" -- "$@") || usage 1
```
[the remark this answers is not in the document]
Comment (kalikiana) [reply script/configure-web-proxy]: https://github.com/os-autoinst/openQA/pull/5769#discussion_r1679584623

**Comment by perlpunk** [comment]
> > I would prefer if the script left `nginx.conf`/`listen.conf` untouched if no port is specified.
> 
> That would lead to surprises when running the script more than once unless we start parsing the original file.

I would agree with @Martchus, leave it untouched if there was no web_port parameter given.
I don't see why it's problematic.

**Review by perlpunk — APPROVED** [review]

File: script/configure-web-proxy [hunk 3]
```diff
@@ -37,10 +43,12 @@ if [[ $web_proxy == "nginx" ]]; then
     echo "Setting up nginx"
     sed "s/openqa.example.com/$(hostname)/" /etc/nginx/vhosts.d/openqa.conf.template > /etc/nginx/vhosts.d/openqa.conf
     sed -i -e "s/\(^[^#]*server_name  localhost;\)/#\1/" /etc/nginx/nginx.conf
+    test -n "$web_port" && sed -i "s/listen.*$/listen $web_port;/" /etc/nginx/nginx.conf
```
Comment (kalikiana) [inline script/configure-web-proxy:46]: `sudo sed -i "s@:80 @:8080 @g" /etc/nginx/vhosts.d/openqa.conf`

Seems like this also needs to be updated, which I noticed after trying this in a new container.

File: docs/Installing.asciidoc [hunk 4]
```diff
@@ -374,8 +374,10 @@ For a local instance setup you can simply execute the script:
 /usr/share/openqa/script/configure-web-proxy
 ----
 
-This will automatically setup a local Apache http proxy. Read on for more
-detailed setup instructions with all the details.
+This will automatically setup a local Apache http proxy. The script
+also supports NGINX and a custom port to listen on. Try `--help` to
```
Comment (okurz) [inline docs/Installing.asciidoc:378]:
```suggestion
also supports nginx and a custom port to listen on. Try `--help` to
```

as written on https://nginx.org/en/

Comment (kalikiana) [reply docs/Installing.asciidoc:378]: We use NGINX in our docs, which is the other option

File: script/configure-web-proxy [hunk 5]
```diff
@@ -36,11 +42,15 @@ sed -i -e 's/^.*httpsonly.*$/httpsonly = 0/g' /etc/openqa/openqa.ini
 if [[ $web_proxy == "nginx" ]]; then
     echo "Setting up nginx"
     sed "s/openqa.example.com/$(hostname)/" /etc/nginx/vhosts.d/openqa.conf.template > /etc/nginx/vhosts.d/openqa.conf
+    sed -i "s@:80 @:$web_port @g" /etc/nginx/vhosts.d/openqa.conf
+    sed -i "s@ 80 @ $web_port @g" /etc/nginx/vhosts.d/openqa.conf
```
[the remark this answers is not in the document]
Comment (kalikiana) [reply script/configure-web-proxy:46]: No. Note the `:`

[the remark this answers is not in the document]
Comment (okurz) [reply script/configure-web-proxy:46]: Why do you use two separate lines then?

```suggestion
    sed -i "s@[: ]80 @:$web_port @g" /etc/nginx/vhosts.d/openqa.conf
```

[the remark this answers is not in the document]
Comment (okurz) [reply script/configure-web-proxy:46]: "unresolved" again as @kalikiana you didn't respond to my message (except for "resolving")

[the remark this answers is not in the document]
Comment (kalikiana) [reply script/configure-web-proxy:46]: Note the : - not sure what else do add here? These are two separate replacements.

File: script/configure-web-proxy (hunk 3 above)
Comment (nicksinger) [reply script/configure-web-proxy:48]: This line seems to duplicate #L45-#L46 no? It also transforms

```
  listen 80;
  listen [::]:80;
```

into 

```
  listen $web_port;
  listen $web_port;
```

which would be an invalid configuration.

**Review by nicksinger — CHANGES_REQUESTED** [review]
I do not care so much about the duplicate sed calls but the call in #L48 needs to be fixed (or just removed?) because it would just break our config.

File: script/configure-web-proxy (hunk 3 above)
Comment (kalikiana) [reply script/configure-web-proxy:48]: It did not work without it. But I guess it might silently break IPv6

**Review by okurz — APPROVED** [review]

File: script/configure-web-proxy [hunk 6]
```diff
@@ -36,11 +42,17 @@ sed -i -e 's/^.*httpsonly.*$/httpsonly = 0/g' /etc/openqa/openqa.ini
 if [[ $web_proxy == "nginx" ]]; then
     echo "Setting up nginx"
     sed "s/openqa.example.com/$(hostname)/" /etc/nginx/vhosts.d/openqa.conf.template > /etc/nginx/vhosts.d/openqa.conf
+    # IPv6 uses [::]:80 syntax instead of just a port
+    if [[ -n "$web_port" ]]; then
+        sed -Ei "s@(.*listen.*)80@\1${web_port}@g" /etc/nginx/vhosts.d/openqa.conf
+        sed -i "s/listen.*$/listen $web_port;/" /etc/nginx/nginx.conf
```
Comment (nicksinger) [inline script/configure-web-proxy:48]: I think we need the same as for the openqa.conf here. WDYT about merging both into one call:
```suggestion
       sed -Ei "s@(.*listen.*)80@\1${web_port}@g" /etc/nginx/vhosts.d/openqa.conf /etc/nginx/nginx.conf
```

**Review by sqozz — APPROVED** [review]

**Review by nicksinger — APPROVED** [review]

**Pull request merged**